### ISSUE TITLE
The task RunVisualStudioTestsusingTestAgent

### ISSUE THREAD

**Issue by kardinal3000** [Microsoft/vsts-tasks#2362]
Hello guys, I had install TFS 15 RC1 and faced with next issue the slicing of test cases coming from single assembly was working (VSO) and were distributed uniformly on two different VMs. But when using on premises build agent ,the slicing is not working. I have already updated build agent to **2.105.2.** I get this warning when run tests task starts.

Executing the powershell script: D:\VSTSAgents\_work\_tasksRunVisualStudioTestsusingTestAgent_d353d6a2-e361-4a8f-8d8c-123bebb71028\1.0.40\RunDistributedTests.ps1
**Update the build agent to run tests with uniform distribution. If you are using hosted agent there are chances that it is still not updated, so retry using your own agent.**

I have  try to use more older version of build agent  103.1, 104.1, 104.2, 105, 105.1 didn't help 

I have managed to produce slicing  on version of Build Agent 103. I mean run distribution tests in parallel mode. But it is not truly parallel mode. I have two test assembly  Test1.dll, Test2.dll  also two test VMs. As I can see, they are running successively. 

On further investigation it appears class InvokeRundistributedTestsCmdLet of  **Microsoft.TeamFoundation.DistributedTask.Task.LegacySDK.dll** doesn't have custom slicing enabled (CustomSlicingEnabled) as a parameter. Why it was working with hosting agent. Does it mean the  Microsoft.TeamFoundationDistributedTask.Task.LegacySDK.dll is different in case of hosted agent ?

Cmdlet  Invoke-RunDistributedTests  15.0.0.0  Microsoft.TeamFoundation.DistributedTask.Task.LegacySDK 
**I can't find  -CustomSlicingEnabled $customSlicingEnabledFlag**

![invoke-rundistributedtests](https://cloud.githubusercontent.com/assets/18549964/18120677/6d811072-6f69-11e6-9d53-9f959f96a09d.png)

**Comment by acesiddhu** [comment]
This one is a duplicate of https://github.com/Microsoft/vsts-tasks/issues/2370.

**Comment by acesiddhu** [comment]
Closing this instance. we will track #2370.
